### ISSUE TITLE
Loading data using dataframe batches duplicates the data

### ISSUE THREAD

**Issue by aecio** [lancedb/lancedb#482]
As explained in the title, loading data using dataframe and a generator leads to duplicating entries. The code below reproduces the issue. Replacing the pandas's `DataFrame` by pyarrow's `RecordBatch` (commented out) yields the expected result.

```python
import numpy as np
import pandas as pd
import lancedb
import pyarrow as pa
import duckdb


def make_batches():
    for i in range(5):
        yield pd.DataFrame({
            "vector": [
                np.array([3.1, 4.1]),
                np.array([5.9, 26.5])
            ],
            "item": ["foo", "bar"],
            "price": [10.0, 20.0],
        })
        # yield pa.RecordBatch.from_arrays(
        #     [
        #         pa.array([np.array([3.1, 4.1]), np.array([5.9, 26.5])], pa.list_(pa.float32(), 2)),
        #         pa.array(["foo", "bar"], pa.utf8()),
        #         pa.array([10.0, 20.0]),
        #     ],
        #     ["vector", "item", "price"],
        # )

schema=pa.schema([
    pa.field("vector", pa.list_(pa.float32(), 2)),
    pa.field("item", pa.utf8()),
    pa.field("price", pa.float32()),
])


db = lancedb.connect("data/lancedb-test/")
tbl = db.create_table("table4", make_batches(), schema=schema)

lance_tbl = tbl.to_lance()

print(duckdb.sql("SELECT * FROM lance_tbl").to_df())
print(duckdb.sql("SELECT COUNT(*) FROM lance_tbl").to_df())
```

```
                                     vector item  price
0   [3.0999999046325684, 4.099999904632568]  foo   10.0
1                 [5.900000095367432, 26.5]  bar   20.0
2   [3.0999999046325684, 4.099999904632568]  foo   10.0
3                 [5.900000095367432, 26.5]  bar   20.0
4   [3.0999999046325684, 4.099999904632568]  foo   10.0
5                 [5.900000095367432, 26.5]  bar   20.0
6   [3.0999999046325684, 4.099999904632568]  foo   10.0
7                 [5.900000095367432, 26.5]  bar   20.0
8   [3.0999999046325684, 4.099999904632568]  foo   10.0
9                 [5.900000095367432, 26.5]  bar   20.0
10  [3.0999999046325684, 4.099999904632568]  foo   10.0
11                [5.900000095367432, 26.5]  bar   20.0
12  [3.0999999046325684, 4.099999904632568]  foo   10.0
13                [5.900000095367432, 26.5]  bar   20.0
14  [3.0999999046325684, 4.099999904632568]  foo   10.0
15                [5.900000095367432, 26.5]  bar   20.0
16  [3.0999999046325684, 4.099999904632568]  foo   10.0
17                [5.900000095367432, 26.5]  bar   20.0
18  [3.0999999046325684, 4.099999904632568]  foo   10.0
19                [5.900000095367432, 26.5]  bar   20.0

   count_star()
0            20
```

`lancedb` version:
```bash
$poetry show lancedb
name         : lancedb
version      : 0.2.2
description  : lancedb

dependencies
 - aiohttp *
 - attr *
 - pydantic *
 - pylance 0.6.5
 - ratelimiter *
 - retry *
 - semver >=3.0
 - tqdm *
```

**Comment by AyushExel** [comment]
Thanks for reporting. This PR should fix it https://github.com/lancedb/lancedb/pull/483

**Comment by AyushExel** [comment]
Thanks for reporting. This PR should fix it https://github.com/lancedb/lancedb/pull/483

**Comment by aecio** [comment]
Looks like the fix has been released in version 0.2.4. Thanks!
